### PR TITLE
enhancement(syntax): add support for variables, classes, and functions

### PR DESCRIPTION
**Pull request by BlackStar8107**
Changes that allow for variables, classes and function to be correctly highlighted!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "author": "Anthony Fuller",
     "publisher": "AnthonyFuller",
     "license": "GPL-3.0-or-later",
-    "version": "0.0.4",
+    "version": "0.0.5",
```
Comment (AnthonyFuller) [inline package.json]: Please do not bump the version for a PR. This should be reverted.

File: .gitignore
Comment (AnthonyFuller) [inline .gitignore:1]: `.mnsc` and `.mnscript` should be added in here too.

File: syntaxes/mnscript.tmLanguage.json [hunk 2]
```diff
@@ -87,6 +149,36 @@
...
+			"end": ";",
+			"beginCaptures": {
+				"0":{ "name":"storage.type.mnscript"}
+			},
+			"patterns": [
+				{
+					"match": "(?<=\\=\\s+)[a-zA-Z\\_0-9]+|[a-zA-Z\\_0-9]+(?=;)|[a-zA-Z\\_0-9]+\\s*(?=\\=)|[a-zA-Z\\_0-9]+\\s*(?!\\))|(?<!\\()[a-zA-Z\\_0-9]|(?<!\\{)[a-zA-Z\\_0-9]",
+					"name": "variable.name.mnscript"
+				}
+			]
```
Comment (AnthonyFuller) [inline syntaxes/mnscript.tmLanguage.json]: I don't recall exactly if the match and begin are strictly necessary, and this bit probably needs a refactor anyway, but that's for another day when the actual language server portion gets implemented.

Respond to the first part of the above sentence, this isn't something that *needs* to be resolved for merging. Just laying out my thoughts in text.

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
Thank you for contributing! Please respond to the comments.

File: syntaxes/mnscript.tmLanguage.json (hunk 2 above)
Comment (BlackStar8107) [reply syntaxes/mnscript.tmLanguage.json]: This does need to be refactored, it's a mess!
I toyed around with removing the begin and end statement, but it comes back to the issue of not highlighting keywords like string correctly, which is why I ended up doing the weird capture to begin with.
Will definitely need to come back and look at this at some point, for now just going to leave it in.

Comment (AnthonyFuller) [reply syntaxes/mnscript.tmLanguage.json]: No problem.

**Pull request merged**